src/sidebar/util/build-thread.js

```diff
@@ -253,38 +227,63 @@ function hasVisibleChildren(thread) {
 }
 
 /**
- * @typedef Options
- * @prop {string[]} selected - List of currently-selected annotation ids, from
- *       the data store
+ * @typedef BuildThreadOptions
+ * @prop {Object.<string, boolean>} expanded - Map of thread id => expansion state
  * @prop {string[]} forcedVisible - List of $tags of annotations that have
  *       been explicitly expanded by the user, even if they don't
  *       match current filters
+ * @prop {string[]} selected - List of currently-selected annotation ids, from
+ *       the data store
  * @prop {(a: Annotation) => boolean} [filterFn] - Predicate function that
  *       returns `true` if annotation should be visible
  * @prop {(t: Thread) => boolean} [threadFilterFn] - Predicate function that
  *       returns `true` if the annotation should be included in the thread tree
- * @prop {Object.<string, boolean>} expanded - Map of thread id => expansion state
- * @prop {(a: Annotation, b: Annotation) => boolean} sortCompareFn - Less-than
- *       comparison function for sorting top-level annotations
- * @prop {(a: Annotation, b: Annotation) => boolean} replySortCompareFn - Less-than
- *       comparison function for sorting replies
+
+ * @prop {(a: Thread, b: Thread) => number} [sortCompareFn] - comparison
+ *       function for sorting top-level annotations
+ * @prop {(a: Thread, b: Thread) => number} [replySortCompareFn] - comparison
+ *       function for sorting replies
+ */
+
+/**
+ * Default top-level thread sorting function if none provided via options.
+ *
+ * Sort by local application-applied `$tag`
+ *
+ * @param {Thread} a
+ * @param {Thread} b
+ * @return {number}
  */
+const defaultSortCompareFn = (a, b) => {
+  if (a.annotation && b.annotation) {
+    if (a.annotation.$tag < b.annotation.$tag) {
+      return -1;
+    } else if (a.annotation.$tag > b.annotation.$tag) {
+      return 1;
+    }
+  }
+  return 0;
+};
 
 /**
- * Default options for buildThread()
+ * Default reply thread sorting function if none provided via options.
  *
- * @type {Options}
+ * Sort by reply (Annotation) `created` date
+ *
+ * @param {Thread} a
+ * @param {Thread} b
+ * @return {number}
  */
-const defaultOpts = {
-  selected: [],
-  expanded: {},
-  forcedVisible: [],
-  sortCompareFn: (a, b) => {
-    return a.$tag < b.$tag;
-  },
-  replySortCompareFn: (a, b) => {
-    return a.created < b.created;
-  },
+const defaultReplySortCompareFn = (a, b) => {
+  if (!a.annotation || !b.annotation) {
+    return 0;
+  }
+  if (a.annotation.created < b.annotation.created) {
+    return -1;
+  } else if (a.annotation.created > b.annotation.created) {
+    return 1;
+  }
```

Comment:
I don't think it should be too hard to have a test that covers these cases? It would be useful to make sure the code executes at least.